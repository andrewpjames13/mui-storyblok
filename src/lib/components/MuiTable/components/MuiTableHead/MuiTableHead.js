import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { TableHead } from '@material-ui/core';
import MuiTableRow from '../MuiTableRow/MuiTableRow';
import StoryBlok from '../../../../utils/Storyblok';

const MuiTableHead = ({
  rootClass,
  content,
}) => {
  const components = {
    MuiTableRow,
  };

  const styles = StoryBlok.arrayToMuiStyles(rootClass);
  return (
    <TableHead className={styles.root}>
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </TableHead>
  );
};

export default MuiTableHead;

MuiTableHead.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /**
   * Content passed to render
   * components:
    MuiTableRow,
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiTableHead.defaultProps = {
  rootClass: [],
};
