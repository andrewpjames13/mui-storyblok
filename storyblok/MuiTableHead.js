module.exports = {
  name: 'MuiTableHead',
  display_name: null,
  updated_at: '2020-04-20T17:01:20.599Z',
  schema: {
    rootClass: {
      type: 'options',
      description: '   * stroyblok multiselect of css classes\n  * Mui Override or extend the styles applied to the component.\n',
    },
    content: {
      type: 'bloks',
      restrict_components: true,
      component_whitelist: [
        'MuiTableRow',
      ],
    },
  },
  image: null,
  preview_field: null,
  is_root: false,
  preview_tmpl: null,
  is_nestable: true,
  all_presets: [],
  preset_id: null,
  real_name: 'MuiTableHead',
  component_group_uuid: null,
};
