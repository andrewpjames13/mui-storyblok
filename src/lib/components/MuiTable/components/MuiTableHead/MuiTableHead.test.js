import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiTableHead from './MuiTableHead';

function setup() {
  const props = {
    content: [
      {
        component: 'MuiTableRow',
        content: [
          {
            component: 'MuiTableCell',
            content: [
              {
                component: 'MuiTypography',
                content: [
                  {
                    component: 'MuiText',
                    text: 'wooo',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  };

  const comp = shallow(<MuiTableHead {...props} />);
  return { comp, props };
}

describe('<MuiTableHead />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiTableHead {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render MuiTableHead', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
