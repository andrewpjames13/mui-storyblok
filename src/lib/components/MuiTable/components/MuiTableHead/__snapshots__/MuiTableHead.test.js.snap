// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiTableHead /> snapshot 1`] = `
<thead
  className="MuiTableHead-root makeStyles-root-2"
  role={null}
>
  <tr
    className="MuiTableRow-root makeStyles-root-4 MuiTableRow-head"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head makeStyles-root-10"
      scope="col"
    >
      <p
        className="MuiTypography-root makeStyles-root-23 MuiTypography-body1"
      >
        wooo
      </p>
    </th>
  </tr>
</thead>
`;
